models/option.go

```diff
@@ -61,6 +94,12 @@ func (option *OptionNode) IsEmpty() bool {
 	return !option.IsValueOption() && !option.IsConfigOption()
 }
 
+// SetType ...
+func (option *OptionNode) SetType(optionType Type) *OptionNode {
```

Comment:
Pls set the type in the Option initializer (`NewOption`) to avoid issues where we do not set type accidentally